applications/asset_tracker_v2/src/cloud/azure_iot_hub_integration.c

```diff
@@ -0,0 +1,329 @@
+#include "cloud/cloud_wrapper.h"
+#include <zephyr.h>
+#include <net/azure_iot_hub.h>
+#include <modem/at_cmd.h>
+
+#define MODULE azure_iot_integration
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_CLOUD_INTEGRATION_LOG_LEVEL);
+
+#if !defined(CONFIG_CLOUD_CLIENT_ID_USE_CUSTOM)
+#define AZURE_IOT_HUB_CLIENT_ID_LEN 15
+#else
+#define AZURE_IOT_HUB_CLIENT_ID_LEN (sizeof(CONFIG_CLOUD_CLIENT_ID) - 1)
+#endif
+
+#define PROP_BAG_COUNT 1
+#define AZURE_IOT_PROP_BAG_BATCH "batch"
+
+#define REQUEST_DEVICE_TWIN_STRING ""
+
+static struct azure_iot_hub_prop_bag prop_bag_batch[PROP_BAG_COUNT] = {
+		[0].key = AZURE_IOT_PROP_BAG_BATCH,
+		[0].value = NULL
+};
+
+static char client_id_buf[AZURE_IOT_HUB_CLIENT_ID_LEN + 1];
+
+static struct azure_iot_hub_config config;
+
+static cloud_wrap_evt_handler_t wrapper_evt_handler;
+
+static void cloud_wrapper_notify_event(const struct cloud_wrap_event *evt)
+{
+	if ((wrapper_evt_handler != NULL) && (evt != NULL)) {
```

Comment:
I don't think we can end up in a situation where this function is called and `wrapper_evt_handler` and `evt` are NULL.
Perhaps this could be changed to an assert on the two, because it would indicate something critically wrong in the underlying libraries if this situation occurs. Then the assert could be traced and reported

Reply:
Agreed. Lets just keep it like this for now since it is done like this in `aws_iot_integration``` as well. We can do the change for all integration layers in the future.